contracts/validator-manager/PoSValidatorManager.sol

```diff
@@ -520,6 +543,18 @@ abstract contract PoSValidatorManager is
             "PoSValidatorManager: delegation not pending added"
         );
 
+        _completeEndDelegation(delegationID);
+    }
+
+    function _completeEndDelegation(bytes32 delegationID) internal {
+        PoSValidatorManagerStorage storage $ = _getPoSValidatorManagerStorage();
+
+        Delegator memory delegator = $._delegatorStakes[delegationID];
+        bytes32 validationID = delegator.validationID;
+
+        // Once this function completes, the delegation is completed so we can clear it from state now.
+        delete $._delegatorStakes[delegationID];
```

Comment:
Can we move this to the end of the function, just ahead of event emission? IMO it would improve readability by making it clearer that the above `Delegator memory` variable is unaffected by the delete.

Reply:
This is to prevent possible reentrancy. We have reentrancy guards, but I think this is good practice to delete the balance/state before calling out to an external function.

Reply:
Makes sense, thanks